Userland/Utilities/less.cpp

```diff
@@ -190,7 +202,52 @@ class Pager {
         return true;
     }
 
+    bool at_end()
+    {
+        return (m_line + m_height - 1) >= m_lines.size() && feof(m_file);
+    }
+
 private:
+    size_t render_status_line(const StringView& prompt, size_t off = 0, char end = '\0', bool ignored = false)
+    {
+        for (; prompt[off] != end && off < prompt.length(); ++off) {
+            if (ignored)
+                continue;
+
+            switch (prompt[off]) {
+            case '?':
+                switch (prompt[++off]) {
+                case 'f':
+                    off = render_status_line(prompt, off + 1, ':', m_file == stdin);
+                    off = render_status_line(prompt, off + 1, '.', m_file != stdin);
+                    break;
+                case 'e':
+                    off = render_status_line(prompt, off + 1, ':', !at_end());
+                    off = render_status_line(prompt, off + 1, '.', at_end());
+                    break;
+                default:
+                    VERIFY_NOT_REACHED();
```

Comment:
what if someone just puts a `?` at the end?
maybe let's not crash?

Reply:
I now treat all unknown guards and specifiers as being false, and treat '?' and '%' at the end of the line as regular characters. I also added  '\' to escape special characters.